packages/auth/esbuild.js

```diff
@@ -0,0 +1,38 @@
+const {build} = require('esbuild')
```

Comment:
I understand esbuild is faster, but for this project it should not matter much.

As such, I'd prefer if we kept it consistent with all other projects in this repo (either all using rollup, or all webpack, or all esbuild) versus different system in each.

Reply:
Consistency is good, but this suggests to stick with rollup because stencil uses it [under the hood](https://stenciljs.com/docs/module-bundling).

However, we are already seeing performance limitations with rollup so I am thinking about the impact of this dependency on the project long-term (2-5 years from now).

Webpack seems like it only gives a [small improvement](https://github.com/coveo/ui-kit/pull/1491).


So, I feel we need to consider transitioning our projects to a next-generation bundler where possible.

Atomic will have to continue using rollup, since it's on Stencil to adjust the bundler.
For the other projects though, I think we can transition.

`@coveo/auth` and `@coveo/bueno` should be easy. This branch shows proof for the former.

The biggest question mark is `@coveo/headless`. There are the 600+ errors to resolve.
My goal is to give headless + esbuild another shot after shipping the saml-client.

I could start auth with rollup and then migrate to esbuild after testing with headless, but I thought of doing the reverse, using it as an opportunity to learn the bundler and see it working.